src/Stack/Setup.hs

```diff
@@ -1085,14 +1091,27 @@ installGHCPosix version downloadInfo _ archiveFile archiveType tempDir destDir =
 
     let runStep step wd env cmd args = do
             menv' <- modifyEnvOverride menv (Map.union env)
-            result <- withWorkingDir wd
-                    $ withEnvOverride menv'
-                    $ withProc cmd args
-                    $ try
-                    . readProcessStdout_
-                    -- Calling the ./configure script requires that stdin is
-                    -- open
-                    . setStdin (useHandleOpen stdin)
+            result <- case logLevel of
+              LevelDebug -> do
+                let logLines = CB.lines .| CL.mapM_ (logInfo . T.decodeUtf8With T.lenientDecode)
+                withWorkingDir wd
```

Comment:
This is good code!  There are some existing utility functions for this, though -  `Stack.Prelude.logProcessStderrStdout`.  This stuff got moved around recently in #3730

Reply:
It appears that this is the only place in the codebase that uses `logProcessStderrStdout`. I'd recommend just removing the function entirely from `Stack.Prelude` and keeping the functionality here. No need to provide general utility functions if they're not of general utility :)

FWIW, this is something I would have done myself in #3730 had I noticed, so good catch to you both.

Reply:
Though, to me this sort of thing seems like a useful thing for a future rio - process.  It is not uncommon to want to involve process output in your log while still keeping your log format.

Reply:
I am leaving `logProcessStderrStdout` as it is being used [here](https://github.com/commercialhaskell/stack/blob/master/src/Stack/Setup.hs#L1320).